### ISSUE TITLE
LibCore: Better error handling for ConfigFile

### ISSUE THREAD

**Issue by sppmacd** [SerenityOS/serenity#9173]
For example, we have:
```c++
    static NonnullRefPtr<ConfigFile> get_for_lib(const String& lib_name);
    static NonnullRefPtr<ConfigFile> get_for_app(const String& app_name);
    static NonnullRefPtr<ConfigFile> get_for_system(const String& app_name);
    static NonnullRefPtr<ConfigFile> open(const String& path);
```
which return `NonnullRefPtr`.
The `void reparse()` just returns when file cannot be opened.

It seems that the only way to check if ConfigFile was properly opened is checking if it has any sections (but ConfigFile without sections is also a valid ConfigFile)

**Comment by guerinoni** [comment]
Mmm you are right but I'm trying to load an empty config file for irc client for example or other applications... In this case there is no crashes or other strange situation, the defaults values are used!
Did you prefer `RefPtr` as return value?

**Comment by sppmacd** [comment]
Yes, I think it would be better if we allow applications handle errors on their own if they want. We can keep default values for simplicity (because that is often used).

We can have two versions of open functions: `get_for_app` which returns `NonnullRefPtr` (and just use default values as currently), and `try_get_for_app` which returns `RefPtr` that is null when open or parsing fails.

**Comment by sin-ack** [comment]
How about `Result<RefPtr<ConfigFile>, String>`? Rich errors are very nice.

**Comment by sppmacd** [comment]
> How about `Result<RefPtr<ConfigFile>, String>`? Rich errors are very nice.

Hmm, why not?

**Comment by sin-ack** [comment]
Oops, meant `Result<NonnullRefPtr<...>, String>`.
